tests/src/test/java/org/eclipse/hono/tests/registry/TenantApiTests.java

```diff
@@ -65,41 +69,63 @@
+        if (tenant.getExtensions() == null) {
+            tenant.setExtensions(new HashMap<>());
+        }
+        tenant.getExtensions().put("customer", "ACME Inc.");
+        Tenants.setAdapterEnabled(tenant, Constants.PROTOCOL_ADAPTER_TYPE_MQTT, true);
+        Tenants.setAdapterEnabled(tenant, Constants.PROTOCOL_ADAPTER_TYPE_HTTP, true, true, httpAdapterExtensions.getMap());
+
+        // expected tenant object
+        final TenantObject expectedTenantObject = TenantObject.from(tenantId, true)
                 .setDefaults(defaults)
-                .setAdapterConfigurations(adapterConfig)
-                .setProperty("customer", "ACME Inc.");
+                .setResourceLimits(new JsonObject()
+                        .put("max-connections", 100000)
+                        .put("ext", new JsonObject()
+                                .put("data-volume", dataVolume)))
+                .setAdapterConfigurations(new JsonArray()
+                        .add(new JsonObject()
+                                .put(TenantConstants.FIELD_ADAPTERS_TYPE, Constants.PROTOCOL_ADAPTER_TYPE_MQTT)
+                                .put(TenantConstants.FIELD_ENABLED, true)
+                                .put(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED, false))
+                        .add(new JsonObject()
+                                .put(TenantConstants.FIELD_ADAPTERS_TYPE, Constants.PROTOCOL_ADAPTER_TYPE_HTTP)
+                                .put(TenantConstants.FIELD_ENABLED, true)
+                                .put(TenantConstants.FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED, true)
+                                .put("ext", httpAdapterExtensions))
+                        );
 
         getHelper().registry
-        .addTenant(JsonObject.mapFrom(tenant))
+        .addTenant(tenantId, tenant)
         .compose(ok -> getAdminClient().get(tenantId))
         .setHandler(ctx.succeeding(tenantObject -> {
             ctx.verify(() -> {
-                assertThat(tenantObject.isEnabled()).isTrue();
-                assertThat(tenantObject.getTenantId()).isEqualTo(tenantId);
-                assertThat(tenantObject.getResourceLimits()).isEqualTo(resourceLimits);
-                assertThat(tenantObject.getDefaults()).isEqualTo(defaults);
-                assertThat(tenantObject.getAdapterConfigurations()).isEqualTo(adapterConfig);
-                assertThat(tenantObject.getProperty("customer", String.class)).isEqualTo("ACME Inc.");
+                assertThat(tenantObject.getResourceLimits()).isEqualTo(expectedTenantObject.getResourceLimits());
+                assertThat(tenantObject.getDefaults()).isEqualTo(expectedTenantObject.getDefaults());
+                assertThat(tenantObject.getAdapterConfigurations()).isEqualTo(expectedTenantObject.getAdapterConfigurations());
+                assertThat(tenantObject.getProperty("ext", JsonObject.class).getString("customer")).isEqualTo("ACME Inc.");
```

Comment:
why don't you simply compare to the values of the *tenant* object that you registered? I do not see why we need to create the *expectedTenantObject*, even more so, since the type of object returned by `getAdminClient().get(String)` is `Tenant`, not `TenantObject`, isn't it?

Reply:
No, `getAdminClient().get(String)` returns a `TenantObject`(specifically, it returns `Future<TenantObject>`.  We use the `Tenant` value object when sending tenant requests and `TenantObject` value object for the response

Reply:
Ah, right, I see.